components/scream/cmake/ScreamUtils.cmake

```diff
@@ -16,6 +16,10 @@ function(CreateUnitTest target_name target_srcs scream_libs)
   set(multiValueArgs MPI_RANKS THREADS CONFIG_DEFS INCLUDE_DIRS)
   cmake_parse_arguments(CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
 
+  if (SCREAM_LIB_ONLY)
```

Comment:
I would not put this check here. We should check the SCREAM_LIB_ONLY to establish whether to include tests subdirs, like you did in p3's main CMakeLists.txt.

Unless you have a reason to keep it, I would remove this.

Reply:
Hey Luca,

I had to add the protection against adding p3/tests since it has some special dependencies set up that won't work when SCREAM_LIB_ONLY is on.

In general, we've done a nice job encapsulating add_executable in this macro, so to me this is a more-elegant solution than chasing down all places that could call this macro.

Reply:
You already added the check on this variable before deciding whether to add p3/tests, so it looks to me as p3 should be fine even without the check inside CreateUnitTest.

Besides, a function signature should be a contract: if you call CreateUnitTest, you _expect_ the test to be created. This is more of a hack.

Adding an `IF(NOT SCREAM_LIB_ONLY)` inside a handful of  CMakeLists.txt (main one, control, dynamics, p3, shoc, rrtmgp) is not such a big 'chase down'.

Reply:
Since the current implementation can lead to confusing behavior, as it did in p3/tests, I've changed my mind and agree with you.